frontend/src/components/course/CourseEditingDialog.js

```diff
@@ -16,39 +16,43 @@ import { useErrorStateValue } from '../../store'
 
 const CourseEditingDialog = ({ state, handleClose, updateCourse, defaultName }) => {
   const [name, setName] = useState('')
+  const [submitDisabled, setSubmitDisabled] = useState(false)
 
   const errorDispatch = useErrorStateValue()[1]
 
   useEffect(() => {
+    if (state.open) {
+      setSubmitDisabled(false)
+    }
     setName(defaultName)
-  }, [defaultName])
+  }, [defaultName, state])
 
-  const handleEdit = async (e) => {
+  const handleEdit = (e) => {
     if (name === '') {
       window.alert('Course needs a name!')
       return
     }
-    try {
-      if (defaultName !== name) {
-        await updateCourse({
-          variables: {
-            id: state.id,
-            name
-          }
+    setSubmitDisabled(true)
+    const shouldUpdate = defaultName !== name
+    const variables = { id: state.id, name }
```

Comment:
Why a separate variable for a short value? I think we don't use a variable like this anywhere else